src/kvstore.c

```diff
@@ -212,6 +220,13 @@ static size_t kvstoreDictMetadataSize(dict *d) {
 /*** API **************************/
 /**********************************/
 
+/* Returns 1 if the corresponding dictionary exists, otherwise returns 0. */
+int kvstoreDictExists(kvstore *kvs, int didx)
```

Comment:
i think we can drop this function and just check `kvstoreGetDict(kvs, didx)!=NULL` where needed

Reply:
It is used in pubsubShardUnsubscribeAllChannelsInSlot to determine whether the dict is deleted

> as you suggested, that the caller checks if the dict still exists (a NULL check, not a size check), and calls the right release function.

originally i used dictSize == 0, i saw this sentence and changed it
https://github.com/redis/redis/pull/13033#issuecomment-1931396771

Reply:
in that use case, dictSize is ok too (no need to keep iterating if the dict got empty).
but we can also call `kvstoreGetDict(kvs, didx)==NULL` if we have it (don't need kvstoreDictExists)

Reply:
> in that use case, dictSize is ok too (no need to keep iterating if the dict got empty).

right, in this case, that's right, i am changing it back.

kvstoreGetDict is a static function, i am removing kvstoreDictExists.